include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -171,6 +172,23 @@ namespace internal
 
 
 
+    template <int dim_to, int dim, int spacedim>
+    std::unique_ptr<FiniteElement<dim_to, dim_to>>
+    create_fe(const FiniteElement<dim, spacedim> &fe)
+    {
+      std::string fe_name = fe.get_name();
+      {
+        const std::size_t template_starts = fe_name.find_first_of('<');
+        Assert(fe_name[template_starts + 1] ==
+                 (dim == 1 ? '1' : (dim == 2 ? '2' : '3')),
+               ExcInternalError());
```

Comment:
I think we rule `FESystem` out somewhere else before, but I think it would still be good if we add an assertion here to check that we do not have an `FESystem`, as that could contain a larger number of `<` symbols and we would create something illegal here.

Reply:
How can I do this? And yes we call in the calling function `const FiniteElement<dim> &fe = fe_in.base_element(base_element_number);` and perform the check if the element is supported in `get_element_type_specific_information()`.

Reply:
I was thinking about something like `Assert(fe_name.find("FESystem") == std::string::npos, ...)` - just to be sure that we do not run into problems here in the future. (Or maybe the downstream error message of `create_fe_from_name` is good enough, it's been a while since I last checked.)